pyodata/v2/service.py

```diff
@@ -684,14 +684,23 @@ class EntityProxy:
 
     # pylint: disable=too-many-branches,too-many-nested-blocks
 
-    def __init__(self, service, entity_set, entity_type, proprties=None, entity_key=None):
+    def __init__(self, service, entity_set, entity_type, proprties=None, entity_key=None, etag=None):
         self._logger = logging.getLogger(LOGGER_NAME)
         self._service = service
         self._entity_set = entity_set
         self._entity_type = entity_type
         self._key_props = entity_type.key_proprties
         self._cache = dict()
         self._entity_key = entity_key
+        self._etag = etag
+
+        if isinstance(proprties, dict):
```

Comment:
You need to document this if statement. Why do you need it? It must be either in the commit message or a comment in the code.

Reply:
The if was there because of that the properties is optional argument, but I now see that is not entirely the case. properties in some cases can be a list of dict! Good Catch!

Reply:
In the case the variable proprties is a list, why don't you handle it? What's is supposed to happen? BTW all the  code below expects the argument proprties holds a dict.

Reply:
But how would you handle it when the list represents multiple entities? Where the ETag would be located? On the first entity? On all of them?

> What's is supposed to happen? BTW all the code below expects the argument proprties holds a dict.

Yes, that it's why it is inside `if isinstance(proprties, dict):`. I don't understand what you are trying to say by it?

Reply:
Why do we have list at first place and how we handle that list anyways?

Reply:
Why do we create an entity proxy but pass a list of properties of multiple entities while one would expect just 1 entity?

Reply:
I think the test_create_entity_nested_list test this behaviour specifically.

Reply:
Hey Martin! You added that if statement, so you should know exactly why you did so. The test you mentioned have a list in a property value which represents a nested entity available via a navigation property. I would say that if we pass a list somewhere, then we have a bug. Please, do your job and exactly describe the case where we pass a list.

Reply:
OK. I had a look into it and the list value is no relevant and is caused by an invalid test - the multiplicity of the navigation property is 1:1, so the service should not respond with a list. You can remove that if statement. We have to add more checks to make sure we detect such an invalid response.

Reply:
Should I update the tests then too?

Reply:
Definitely, you must. Please, in a separate commit which should be place before the current commit.

Reply:
How is this going? Would love to help push this out to release soon as the part of our app that uses the pyodata module is ready to be deployed!

Reply:
The if statement has been removed as there was a bug in a test case which is fixed by:
https://github.com/SAP/python-pyodata/pull/116/commits/079ae2020c85929d732f0daa59f6e21bb3a1f78d